static/js/search_suggestion.js

```diff
@@ -162,6 +162,80 @@ function get_private_suggestions(all_people, operators, person_operator_matches)
     return suggestions;
 }
 
+function get_group_suggestions(all_people, operators) {
+    if (operators.length === 0) {
+        return [];
+    }
+
+    if ((operators[0].operator === 'is') && (operators[0].operand === 'private')) {
+        operators = operators.slice(1);
+    }
+
+    if (operators.length !== 1 || operators[0].operator !== 'pm-with') {
+        return [];
+    }
+
+    var operand = operators[0].operand;
+    var negated = operators[0].negated;
+
+    // The operand has the form "part1,part2,pa", where all but the last part
+    // are emails, and the last part is an arbitrary query.
+    //
+    // We only generate group suggestions when there's more than one part, and
+    // we only use the last part to generate suggestions.
+    var all_but_last_part;
+    var last_part;
+
+    var last_comma_index = operand.lastIndexOf(',');
+    if (last_comma_index < 0) {
+        return [];
+    }
+
+    // Neither all_but_last_part nor last_part include the final comma.
+    all_but_last_part = operand.slice(0, last_comma_index);
+    last_part = operand.slice(last_comma_index + 1);
+
+    // We don't suggest a person if their email is already present in the
+    // operand (not including the last part).
+    var parts = all_but_last_part.split(',');
+    var is_already_present = {};
+    var i;
+    for (i = 0; i < parts.length; i += 1) {
+        is_already_present[parts[i]] = true;
+    }
```

Comment:
We use the underscorejs.org which has a lot of handy helpers for things like this; I think `_.contains()` could work here